mnefun/_mnefun.py

```diff
@@ -1419,7 +1419,21 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
             evokeds = list()
             for name in names:
                 if p.average_weighting == 'chpi':
-                    weights = _chpi_snr_epochs(e[name])
+                    # read in raw unfiltered files for cHPI weighting evokeds
+                    raw_names = get_raw_fnames(p, subj, 'raw', False, True,
+                                               run_indices[si])
+                    raw = [Raw(fname, preload=False, allow_maxshield=True)
+                           for fname in raw_names]
+                    raw_unfil = concatenate_raws(raw)
+                    eps_unfil = Epochs(raw_unfil, events=e[name].events,
+                                       event_id=None, tmin=p.tmin,
+                                       tmax=p.tmax, baseline=_get_baseline(p),
+                                       reject=None, flat=None,
+                                       proj=None, preload=True,
+                                       decim=decim[si], picks=e.picks,
+                                       on_missing='warning')
+                    # Use unfiltered epochs for weights
```

Comment:
What we want is unprocessed data, preferably straight from the MEG. For example, lowpass filtering could remove the cHPI signals. I'm not sure what @Eric89GXL means by decimation above; if the raw data is downsampled before feeding it to chpi_snr, the new sampling freq must be high enough to capture the cHPI signals.